WalletWasabi/WabiSabi/Backend/PostRequests/InputRegistrationHandler.cs

```diff
@@ -127,6 +127,11 @@ public static class InputRegistrationHandler
 				throw new WabiSabiProtocolException(WabiSabiProtocolErrorCode.WrongPhase);
 			}
 
+			if (round.WeightCredentialIssuer.Balance + zeroWeightCredentialRequests.Delta > config.MaxCoinJoinTransactionWeight)
```

Comment:
ZeroCredentialsRequests.Delta is always 0.

Reply:
Currently the `CredentialIssuer` checks for the lower-bound limit and i think it should also enforce the upper-bound limit because that guaranties consistency in all kind of credentials, not only for weight.

Reply:
this should check that `round.WeightCredentialIssuer.Balance + RegistrableWeightCredentials` does not exceed the value.

At this point in the flow the coordinator also sees the prevouts and can calculate the credential amount that should be requested, so this value should be saved in the Alices collection to ensure `realWeightCredentialRequests.Delta` matches later when Alice confirms the connection.

Reply:
if it is zero at this point then it makes no sense to check it here.

Reply:
the credential delta is zero, but the input amount is not, and should match that value exactly.

if two alices try to register and there's only room for one of those the input registration request should fail for one of them, it shouldn't take until connection confirmation

Reply:
Added it back.